llvm/lib/Analysis/IVDescriptors.cpp

```diff
@@ -900,6 +993,11 @@ bool RecurrenceDescriptor::isReductionPHI(PHINode *Phi, Loop *TheLoop,
                       << *Phi << "\n");
     return true;
   }
+  if (AddReductionVar(Phi, RecurKind::IFindLastIV, TheLoop, FMF, RedDes, DB, AC,
```

Comment:
Good question. We don't need to call `AddReductionVar` again for FFindLastIV.
This is because at the end of function `isFindLastIVPattern`, IFindLastIV can be transformed into FFindLastIV if the predicate is an fcmp instruction.

Reply:
It looks like https://github.com/llvm/llvm-project/pull/118393 is attempting to improve this issue.